File: Second-Largest-Digit-in-a-String.py
1class Solution:
2    def secondHighest(self, s: str) -> int:
3        digits = list()
4        for ch in s:
5            if(ch.isdigit()):
6                digits.append(int(ch))
7        fmax,smax = -1,-1
8        for index in range(len(digits)):
9            if(digits[index] > fmax):
10                smax = fmax
11                fmax = digits[index]
12            elif(digits[index] < fmax and digits[index] > smax):
13                smax = digits[index]
14        return smax
15        

File: Second-Largest-Digit-in-a-String.cpp
1class Solution {
2public:
3    int secondHighest(string s) {
4        vector<int> digits;
5        for(char ch : s){
6            if(isdigit(ch)){
7                digits.push_back(ch-'0');
8            }
9        }
10        int fmax = INT_MIN, smax = INT_MIN;
11        for(int num : digits){
12            if(num > fmax){
13                smax = fmax;
14                fmax = num;
15            }
16            else if(num > smax && num < fmax){
17                smax = num;
18            }
19        }
20        return (smax!=INT_MIN) ? smax : -1;
21    }
22};